File: FlowPathwayDiagram/FlowPathwayLine.vb
'************************************************************************************
' STSimStockFlow: A .NET library for simulating stocks and flows
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Drawing
Imports SyncroSim.Common.Forms

Class FlowPathwayLine
    Inherits ConnectorLine

    Private m_Pathway As FlowPathway

    Public Sub New(ByVal lineColor As Color, ByVal pathway As FlowPathway)

        MyBase.New(lineColor)
        Me.m_Pathway = pathway

    End Sub

    Public ReadOnly Property Pathway As FlowPathway
        Get
            Return Me.m_Pathway
        End Get
    End Property

End Class

File: src/Diagram/FlowPathwayLine.cs
// stsim-stockflow: SyncroSim Add-On Package (to stsim) for integrating stocks and flows into state-and-transition simulation models in ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System.Drawing;
using SyncroSim.Common.Forms;

namespace SyncroSim.STSimStockFlow
{
	internal class FlowPathwayLine : ConnectorLine
	{
		private FlowPathway m_Pathway;

		public FlowPathwayLine(Color lineColor, FlowPathway pathway) : base(lineColor)
		{
			this.m_Pathway = pathway;
		}

		public FlowPathway Pathway
		{
			get
			{
				return this.m_Pathway;
			}
		}
	}
}